Add dockerFile option

cgrimal opened the pull request:
Define the `dockerFile` option that is mutually exclusive with the `dockerImage` option.

If `dockerFile` is provided, it must be the name of the Dockerfile file, placed in the current directory.
When building requirements, a Docker image is built based on the provided Dockerfile, and requirements are then built using this image.

**Limitations:**
- built every time (even if image exists and Dockerfile has not changed)
- Dockerfile can have any name but MUST be in the current directory

**Open Questions:**
- If `dockerizePip` is false but any docker related options are provided, no warning are raised. Should I fix it?
- Are these limitations an issue? Do you have ideas how to improve on them?

**Related issues:**
- Closes #114
- Closes #126
- Closes #101 
- Closes #103

dschep commented on lib/pip.js:
```diff
@@ -53,7 +53,32 @@ function installRequirements() {
-    this.serverless.cli.log(`Docker Image: ${this.options.dockerImage}`);
+    let dockerImage = this.options.dockerImage;
+    if (this.options.dockerFile) {
+      this.serverless.cli.log(`Building custom docker image from ${this.options.dockerFile}...`);
```
Could you move this docker build logic to a new `lib/docker.js` module?

cgrimal commented:
Thanks for your feedback @dschep!

- It is indeed possible to place the Dockerfile in a subfolder. I updated the README. That said, you can't have it elsewhere, or Docker will raise a "The Dockerfile must be within the build context (.)" error. FYI, A working solution in command-line is to do `docker build -t sls-py-reqs-custom - < /path/to/Dockerfile` but I can't make it work with `spawnSync`. I tried using the `input` option but I couldn't make it work.

- I added a warning (not exiting the process) if one of the docker related option is provided with `dockerizePip` set to false.

- I moved some of the docker logic to a dedicated `lib/docker.js` module.

Let me know what you think.
